src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -248,10 +250,13 @@ private void createExecute(WorkflowRequest request, User user, ActionListener<Wo
                                             ActionListener.wrap(stateResponse -> {
                                                 logger.info("Creating state workflow doc: {}", globalContextResponse.getId());
                                                 if (request.isProvision()) {
+                                                    String waitForTimeCompletion = request.getParams()
+                                                        .getOrDefault(WAIT_FOR_COMPLETION_TIMEOUT, TimeValue.MINUS_ONE.toString());
```

Comment:
In the case this param doesn't exist (probably most of the time), we shouldn't convert this to a string just to parse it back.   This is probably a good opportunity for a `containsKey()` conditional.